File: KITTY/KURS_1/JAVA/DAY_29_3_FRAGMRNT/app/src/main/java/com/example/velic/day_29_3_fragmrnt/SecondActivity.java
package com.example.velic.day_29_3_fragmrnt;

import android.app.Activity;
import android.app.FragmentManager;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.util.Log;

public class SecondActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d("DAY_29_3_11", "SecondActivity : ");
        setContentView(R.layout.activity_second);
        Log.d("DAY_29_3_11A", "SecondActivity : ");
        if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
            Log.d("DAY_29_3_12", "SecondActivity : LANDSCAPE");
            finish();
            Log.d("DAY_29_3_13", "SecondActivity : LANDSCAPE");
            return;
        }
        Log.d("DAY_29_3_14", "SecondActivity : NON LANDSCAPE");
        // Получим индекс из намерения активности
        Intent intent = getIntent();
        int buttonIndex = intent.getIntExtra("buttonIndex", -1);
        if (buttonIndex != -1) {
            Log.d("DAY_29_3_15", "SecondActivity : NON LANDSCAPE "+buttonIndex);
            FragmentManager fragmentManager = getFragmentManager();
            Fragment2 fragment2 = (Fragment2) fragmentManager
                    .findFragmentById(R.id.fragment2);
            fragment2.setDescription(buttonIndex);
        }
    }
}

File: KITTY/KURS_1/KOTLIN/DAY_29_3_FragmentSecond_kt/app/src/main/java/com/example/day_29_3_fragmentsecond_k/SecondActivity.kt
package com.example.day_29_3_fragmentsecond_k

import android.content.res.Configuration
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        if (resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE) { finish() ; return }

        // Получим индекс из намерения активности
        val intent = intent
        val buttonIndex = intent.getIntExtra("buttonIndex", -1)
        if (buttonIndex < 0)  return

            val fragmentManager = supportFragmentManager
            val fragment2 = fragmentManager
                    .findFragmentById(R.id.fragment2) as Fragment2?
            fragment2?.setDescription(buttonIndex)
    }
}

// Остался один штрих.
// Если в портретной ориентации перейти на вторую активность
// и повернуть устройство в альбомную ориентацию,
// то увидим не тот результат, на который рассчитывали.
// Мы ожидаем увидеть двухпанельный режим, а видим альбомную ориентации второй активности.
// Добавим в метод onCreate() второй активности проверку текущего состояния ориентации.
// Если обнаружим, что находимся в альбомной ориентации, то просто выходим из активности.
// Конечно, можно дополнительно запоминать, какая кнопка была нажата,
// чтобы при возврате на первую активность вывести соответствующие картинки,
// но это уже детали.